CMakeLists.txt

```diff
@@ -196,7 +202,6 @@ if (NOT MSVC)
   if (NOT protobuf_HAVE_BUILTIN_ATOMICS)
     set(protobuf_LINK_LIBATOMIC true)
   endif (NOT protobuf_HAVE_BUILTIN_ATOMICS)
-  set(CMAKE_REQUIRED_FLAGS ${OLD_CMAKE_REQUIRED_FLAGS})
 endif (NOT MSVC)
```

Comment:
Looking at the history, I think this entire atomic check comes from when we supported pre-C++11 and is severely outdated.    It seems safe to assume that std::atomic is present in C++17 and newer, feel like some opportunistic cleanup?

Reply:
Otherwise I'd ask that you leave the tech-debt to make it clear that this can be removed.  The lines you removed were what made me think of it

Reply:
I have removed this check.